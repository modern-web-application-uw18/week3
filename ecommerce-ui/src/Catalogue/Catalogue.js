import React, { Component } from 'react';
import Unit from '../Unit/Unit.js';
import PropTypes from 'prop-types';

class Catalogue extends Component {
    render() {        
        const units = this.props.inventory.map((unit, idx) => {
            return (
                <div className="Unit" key={idx}>
                    <Unit 
                        unit={unit}
                        onAddToCart={this.props.onAddToCart}
                    />
                </div>
            );
        });
        return (
            <div className="Catalogue"> {units} </div>
        );
    }
}

Catalogue.propTypes = {
    inventory: PropTypes.array,
    onAddToCart: PropTypes.func
}

export default Catalogue;