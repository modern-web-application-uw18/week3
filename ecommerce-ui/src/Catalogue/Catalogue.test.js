import React from 'react';
import ReactDom from 'react-dom';
import TestRenderer from 'react-test-renderer';
import Catalogue from './Catalogue.js'

describe('Catalogue', () => {
    it('should clear smoke test for Catalogue rendering', () => {
        const div = document.createElement('div');
        const dummyFunc = (something) => {
            return () => {
                this.setState((prevState, props) => {
                    const purchasedUnit = something;
                    const newPurchasedUnits = prevState.purchasedUnits;
                    newPurchasedUnits.push(purchasedUnit);
                    return {
                        purchasedUnits: newPurchasedUnits
                    };
                });
            }
        };
        const dummyProp = [
            {
                "title": "Cardboard Box",
                "houseType": "Apartment",
                "image": "./itsABox.jpg",
                "location": {
                    "city": "Seattle",
                    "country": "USA"
                },
                "payment": {
                    "cost": 1,
                    "description": "Free cancellation"
                },
                "host": {
                    "name": "I5 underpass",
                    "isSuperhost": false
                },
                "rating": {
                    "stars": 1,
                    "reviews": 6876
                }
            }
        ];
        ReactDom.render(<Catalogue inventory={dummyProp} onAddToCart={dummyFunc}/>, div);
        ReactDom.unmountComponentAtNode(div);
    });

    it('should render a minimal catalogue section of store front', () => {
        const dummyFunc = (something) => {
            return () => {
                this.setState((prevState, props) => {
                    const purchasedUnit = something;
                    const newPurchasedUnits = prevState.purchasedUnits;
                    newPurchasedUnits.push(purchasedUnit);
                    return {
                        purchasedUnits: newPurchasedUnits
                    };
                });
            }
        };
        const dummyProp = [
            {
                "title": "Cardboard Box",
                "houseType": "Apartment",
                "image": "./itsABox.jpg",
                "location": {
                    "city": "Seattle",
                    "country": "USA"
                },
                "payment": {
                    "cost": 1,
                    "description": "Free cancellation"
                },
                "host": {
                    "name": "I5 underpass",
                    "isSuperhost": false
                },
                "rating": {
                    "stars": 1,
                    "reviews": 6876
                }
            }
        ];
        const component = TestRenderer.create(<Catalogue inventory={dummyProp} onAddToCart={dummyFunc}/>);
        expect(component.toJSON()).toMatchSnapshot();
    });
})