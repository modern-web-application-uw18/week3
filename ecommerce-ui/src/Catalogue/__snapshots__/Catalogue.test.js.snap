// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Catalogue should render a minimal catalogue section of store front 1`] = `
<div
  className="Catalogue"
>
   
  <div
    className="Unit"
  >
    <div>
      <p>
        Cardboard Box
      </p>
      <img
        alt="a house you can rent"
        src="./itsABox.jpg"
      />
      <p>
        Location: 
        Seattle
        , 
        USA
      </p>
      <div
        className="PurchaseSection"
      >
        <p>
          Cost: $
          1
        </p>
        <button
          onClick={[Function]}
        >
          Add to Cart
        </button>
      </div>
    </div>
  </div>
   
</div>
`;
